### PR TITLE
Bring API up-to-date with DM v2 documentation

### LINKED ISSUE #189: Update JSON schema to match Data Model v2
https://github.com/scitran/core/wiki/Data-Model,-v2

Changes:
- [x] Rename all `metadata` fields to `info`
  - [x] Rename `info` variable in `placer.py` and `upload.py` to file_attrs to avoid confusion between field and variable. `metadata` now only refers to data sent by client as part of upload process, `info` refers to the file object field representing a key-value store, `file_attrs` refers to the python variable that passes file object attributes persisted to the DB
- [x] Remove acquisition `instrument` and `measurement` fields
- [x]  `file.instrument` becomes `file.modality`
- [x] Ensure all italicized fields are returned in all GET requests (list and singular)
- [x] Return entire permissions list for admin GET requests that return lists

### PR DESCRIPTION
**Pull request by nagem**
Closes #189, #379

List of changes available in issue ticket linked above. Still need to address "Ensure all italicized fields are returned in all GET requests (list and singular)", but since this is a large change, getting eyes on it and testing done sooner rather than later will help.

### CODE REVIEW

File: test/unit_tests/test_rules.py [hunk 1]
```diff
@@ -82,17 +82,6 @@ def test_eval_match_file_measurements():
     result = rules.eval_match(*args)
     assert result == False
 
-def test_eval_match_container_measurement():
```
Comment (rentzso) [inline test/unit_tests/test_rules.py]: Is it ok to remove this test?

File: bin/database.py [hunk 2]
```diff
@@ -79,6 +79,55 @@ def update_file_origins(cont_list, cont_name):
...
+    def dm_v2_updates(cont_list, cont_name):
+        for container in cont_list:
+            
+            query = {'_id': container['_id']}
+            update = {'$rename': {'metadata': 'info'}}
+
+            files = container.get('files', None)
+            if files is not None:
+                updated_files = []
+                for file in files:
```
Comment (rentzso) [inline bin/database.py]: `file` is a reserved word in python. It's better to use `file_` or `f`.

File: test/unit_tests/test_rules.py (hunk 1 above)
Comment (nagem) [inline test/unit_tests/test_rules.py]: I believe so as it's testing that the file's measurement field matches the acquisition's, and acquisition lost its measurement field.

File: bin/database.py [hunk 3]
```diff
@@ -79,6 +79,55 @@ def update_file_origins(cont_list, cont_name):
...
+    Acquisition fields `instrument` and `measurement` removed 
+    """
+
+    def dm_v2_updates(cont_list, cont_name):
+        for container in cont_list:
+            
+            query = {'_id': container['_id']}
+            update = {'$rename': {'metadata': 'info'}}
+
+            files = container.get('files', None)
```
Comment (rentzso) [inline bin/database.py]: No need to specify the default value here.
The `get` method of a python dict will return `None` if it doesn't find the key anyway.

Please also check the other lines below with default value `None`

File: test/unit_tests/test_rules.py (hunk 1 above)
Comment (rentzso) [inline test/unit_tests/test_rules.py]: I see. Maybe we should remove also lines 87-89 in `rules.py` then.

File: api/rules.py [hunk 4]
```diff
@@ -84,10 +82,6 @@ def eval_match(match_type, match_param, file_, container):
             _log_file_key_error(file_, container, 'has no measurements key')
             return False
 
-    # Match the container's primary measurment
```
Comment (nagem) [inline api/rules.py]: @rentzso In reference to your comment on test_rules like we already caught it.

File: bin/database.py [hunk 5]
```diff
@@ -79,6 +79,55 @@ def update_file_origins(cont_list, cont_name):
...
+                    {'metadata': { '$exists': True}}]}
+
+    dm_v2_updates(config.db.collections.find(query), 'collections')
+    dm_v2_updates(config.db.projects.find(query), 'projects')
+    dm_v2_updates(config.db.sessions.find(query), 'sessions')
+
+    query['$or'].append({'instrument': { '$exists': True}})
+    query['$or'].append({'measurement': { '$exists': True}})
+    dm_v2_updates(config.db.acquisitions.find(query), 'acquisitions')
+
```
Comment (rentzso) [inline bin/database.py]: For performance it could be beneficial to check [Mongodb Aggregation framework](https://docs.mongodb.org/manual/aggregation/)

**Comment by rentzso** [comment]
LGTM pending comments

**Comment by nagem** [comment]
Rebased.

**Comment by nagem** [comment]
Comment from PR added, this is officially ready to merge but is a breaking change.

**Comment by kofalt** [comment]
I'm afraid that this has conflicts again.

**Comment by kofalt** [comment]
Assigning to @dpuccetti based on team discussion - please link/create relevant frontend ticket as a blocker for this ticket.

**Comment by nagem** [comment]
@rentzso  you can see the changes in the schemas for the upload endpoints [here](https://github.com/scitran/core/pull/246/files#diff-cfbbcb154b7b2acc5522873b1dd365a8). I plan on making the changes to the wiki entry soon, too. Let me know if you need any more specifics!

**Pull request merged**